examples/spmm/spmm.cc

```diff
@@ -195,14 +195,16 @@ class Write_SpMatrix : public Op<Key<2>, std::tuple<>, Write_SpMatrix<Blk>, Blk>
       : baseT(edges(in), edges(), "write_spmatrix", {"Cij"}, {}, [](auto key) { return 0; }), matrix_(matrix) {}
 
   void op(const Key<2> &key, typename baseT::input_values_tuple_type &&elem, std::tuple<> &) {
-    std::lock_guard<std::mutex> lock(mtx_);
+    mtx_.lock();
     if (ttg::tracing()) {
       auto &w = get_default_world();
       ttg::print(w.rank(), "/", reinterpret_cast<std::uintptr_t>(pthread_self()), "spmm.impl.h Write_SpMatrix wrote {",
                  key[0], ",", key[1], "} = ", baseT::template get<0>(elem), " in ", static_cast<void *>(&matrix_),
                  " with mutex @", static_cast<void *>(&mtx_), " for object @", static_cast<void *>(this));
     }
-    matrix_.insert(key[0], key[1]) = baseT::template get<0>(elem);
+    auto entry = matrix_.insert(key[0], key[1]);
+    mtx_.lock();
+    entry = baseT::template get<0>(elem);
```

Comment:
For posterity: `entry` can be invalidated by inserts happening since lock release (if such insert triggers internal buffer resizing)

Reply:
Mhh, that's unfortunate. That makes the critical path extremely much longer... Any idea how to work around this?

Reply:
should not be critical for shallow copy tiles, #120